### PR TITLE
Added one of the two commands, the shooter intake command

### LINKED ISSUE #78: Intake from source command
Raise the ShooterPivot to a degree to be able to intake from the source

Run the shooter and intake backwards until beam-break sensor is broken.

### PR DESCRIPTION
**Pull request by PureBreadCat**
fixes #78

### CODE REVIEW

File: robotbase/src/main/java/frc/robot/commands/pickup/ShooterIntakeFromSource.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import frc.robot.Robot;
+
+public class ShooterIntakeFromSource extends Command{
+    public ShooterIntakeFromSource(){
+        addRequirements(Robot.intake,Robot.shooter,Robot.pivot);
+    }
+
+    @Override
+    public void initialize(){
+        Robot.pivot.setPivotRotations(Constants.SHOOTER_PIVOT.INTAKE_FROM_SOURCE_ROTATIONS);
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/frc/robot/commands/pickup/ShooterIntakeFromSource.java:14]: This should be done using a parallel command group with commands specifically for each subsystem. We should never directly interact with multiple subsystems in a single command.

**Review by NolanCampbell2357 — CHANGES_REQUESTED** [review]

File: robotbase/src/main/java/frc/robot/Constants.java [hunk 2]
```diff
@@ -137,6 +137,9 @@ public static final class CONTROLLER {
   }
 
   public static final class SHOOTER {
+    public static final double TOP_MOTOR_PICKUP_SPEED_RPMS = 0;
+    public static final double BOTTOM_MOTOR_PICKUP_SPEED_RPMS = 0;
```
Comment (Tyson453) [inline robotbase/src/main/java/frc/robot/Constants.java:141]: rename to `TOP_MOTOR_SOURCE_INTAKE_RPMS`

File: robotbase/src/main/java/frc/robot/Constants.java [hunk 3]
```diff
@@ -167,6 +170,9 @@ public static final class INTAKE {
     public static final double TOP_MOTOR_PICKUP_SPEED_PERCENT_OUTPUT = 0;
     public static final double BOTTOM_MOTOR_PICKUP_SPEED_PERCENT_OUTPUT = 0;
 
+    public static final double TOP_MOTOR_EAT_SPEED_PERCENT_OUTPUT = 0;
+    public static final double BOTTOM_MOTOR_EAT_SPEED_PERCENT_OUTPUT = 0;
```
Comment (Tyson453) [inline robotbase/src/main/java/frc/robot/Constants.java:174]: rename to `TOP_MOTOR_SOURCE_INTAKE_SPEED_PERCENT_OUTPUT`

File: robotbase/src/main/java/frc/robot/commands/intake/IntakeNoteFromSource.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package frc.robot.commands.intake;
```
Comment (Tyson453) [inline robotbase/src/main/java/frc/robot/commands/intake/IntakeNoteFromSource.java:1]: Rename command to `IntakeReverseFeed`

File: robotbase/src/main/java/frc/robot/commands/pivot/ShooterPivotIntakeFromSource.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package frc.robot.commands.pivot;
```
Comment (Tyson453) [inline robotbase/src/main/java/frc/robot/commands/pivot/ShooterPivotIntakeFromSource.java:1]: This is fine for now. Eventually we'll have a command that will just set the rotations that are passed into the command (I haven't done that yet).

File: robotbase/src/main/java/frc/robot/commands/pivot/ShooterPivotIntakeFromSource.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  }
+
+  @Override
+  public void initialize() {
+    Robot.pivot.setPivotRotations(Constants.SHOOTER_PIVOT.INTAKE_FROM_SOURCE_ROTATIONS);
+  }
+
+  @Override
+  public boolean isFinished() {
+    return false;
```
Comment (Tyson453) [inline robotbase/src/main/java/frc/robot/commands/pivot/ShooterPivotIntakeFromSource.java:19]: `return Robot.pivot.isPivotAtRotations()`

File: robotbase/src/main/java/frc/robot/commands/pickup/ShooterIntakeFromSource.java [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package frc.robot.commands.pickup;
```
Comment (Tyson453) [inline robotbase/src/main/java/frc/robot/commands/pickup/ShooterIntakeFromSource.java:1]: Rename to `SourceIntakeFromShooter`

File: robotbase/src/main/java/frc/robot/commands/shooter/ShooterPickupFromSource.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
+package frc.robot.commands.shooter;
```
Comment (Tyson453) [inline robotbase/src/main/java/frc/robot/commands/shooter/ShooterPickupFromSource.java:1]: This is fine for now. Eventually we'll have a command that sets the RPMs (I haven't done that yet). I'll fix this in my branch.

**Review by Tyson453 — CHANGES_REQUESTED** [review]

File: robotbase/src/main/java/frc/robot/Constants.java [hunk 9]
```diff
@@ -137,6 +137,9 @@ public static final class CONTROLLER {
   }
 
   public static final class SHOOTER {
+    public static final double TOP_MOTOR_SOURCE_INTAKE_RPMSS = 0;
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/frc/robot/Constants.java:140]: Extra S

File: robotbase/src/main/java/frc/robot/commands/intake/IntakeReverseFeed.java [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+public class IntakeReverseFeed extends Command {
+
+  public IntakeReverseFeed() {
+    addRequirements(Robot.intake);
+  }
+
+  @Override
+  public void initialize() {
+    Robot.intake.set(
+        -Constants.INTAKE.TOP_MOTOR_SOURCE_INTAKE_SPEED_PERCENT_OUTPUT,
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/frc/robot/commands/intake/IntakeReverseFeed.java:16]: Pullin the constants file like this `frc.robot.Constants.*` so you do not need toreference `Constant` everytime.

**Review by NolanCampbell2357 — CHANGES_REQUESTED** [review]

**Review by NolanCampbell2357 — APPROVED** [review]

**Pull request merged**